### PR TITLE
Ruby: add meta-queries for measuring sources, sinks, and call edges

### PR DESCRIPTION
**Pull request by asgerf**
Adds three meta-queries for measuring the number of taint sources, sinks, and call edges.

The idea is to include these in a DCA evaluation (as meta-queries) so it's possible to see the impact of your changes at a more fine-grained level. Often, a change generates no new alerts, but produces sources or sinks that give an indication of its impact. The idea was initially discussed in [this issue](https://github.com/github/codeql-javascript-team/issues/191). As for JS, I didn't include logging-related sinks as they tend to dominate too much compared to their security-relevance.

As an example, [this diff](https://github.com/github/codeql-dca-main/blob/data/asgerf/meta-queries__Differences__smoke-test/reports/alert-meta-comparison.md) is comparing `main` to some earlier version from about a week ago, showing plenty of new sinks.

### CODE REVIEW

File: ruby/ql/src/queries/meta/internal/TaintMetrics.qll [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+RemoteFlowSource relevantTaintSource(string kind) {
+  result.getLocation().getFile() instanceof RelevantFile and
+  kind = result.getSourceType()
+}
+
+DataFlow::Node relevantTaintSink(string kind) {
+  result.getLocation().getFile() instanceof RelevantFile and
+  (
+    kind = "CodeInjection" and result instanceof CodeInjection::Sink
```
Comment (nickrolfe) [inline ruby/ql/src/queries/meta/internal/TaintMetrics.qll:24]: Any reason not to include `CleartextLogging::Sink`?

**Review by nickrolfe — COMMENTED** [review]
Thanks for doing this.

I have one question, but otherwise it looks good.

File: ruby/ql/src/queries/meta/internal/TaintMetrics.qll (hunk 1 above)
Comment (asgerf) [reply ruby/ql/src/queries/meta/internal/TaintMetrics.qll:24]: From the PR description,
> As for JS, I didn't include logging-related sinks as they tend to dominate too much compared to their security-relevance.

If you want to measure the number of logging calls, I'd probably put them in a separate meta-query.

Comment (nickrolfe) [reply ruby/ql/src/queries/meta/internal/TaintMetrics.qll:24]: Sorry for not reading more closely.

**Review by nickrolfe — APPROVED** [review]

**Pull request merged**